### DISCUSSION TITLE
Is there a way to specify --remote for submodule update?

### DISCUSSION THREAD

**Discussion by sheaad01** [actions/checkout#638] [Q&A]
Title says it all. I have a submodule that is updated frequently, so keeping the HEAD value up to date in the superproject is a pain. The easy workaround is to specify --remote when performing the submodule checkout. I've read the documentation and searched the web, but haven't found anything. Does the action support this?

**Comment by jpluscplusm** [comment]
+1 for needing this.

The only workaround I can see right now is manually running `git submodule update --remote` as a step, following each&every `actions/checkout` steps.

**Reply by sheaad01** [reply]
That's exactly what I'm doing right now since the action doesn't appear to support what we need.

**Comment by fxnn** [comment]
According to [git-submodule docs](https://git-scm.com/docs/git-submodule#Documentation/git-submodule.txt---remote), another workaround besides manually running `git submodule update --remote` would be a `git pull` inside the submodule.
